packages/core-components/src/components/EmptyState/MissingAnnotationEmptyState.tsx

```diff
@@ -61,19 +46,38 @@ const useStyles = makeStyles<BackstageTheme>(
   { name: 'BackstageMissingAnnotationEmptyState' },
 );
 
-function generateLineNumbers(lineCount: number) {
-  return Array.from(Array(lineCount + 1).keys(), i => i + ANNOTATION_LINE);
-}
+function generateYamlExample(
+  annotations: string[],
+  entity?: Entity,
+): { yamlText: string; lineNumbers: number[] } {
+  const kind = entity?.kind || 'Component';
+  const name = entity?.metadata.name || 'example';
+  const type = entity?.spec?.type || 'website';
+  const owner = entity?.spec?.owner || 'user:default/guest';
 
-function generateComponentYaml(annotations: string[]) {
-  const annotationYaml = annotations
-    .map(ann => ANNOTATION_YAML.replace('ANNOTATION', ann))
-    .join('\n');
+  const yamlText = `apiVersion: backstage.io/v1alpha1
+kind: ${kind}
+metadata:
+  name: ${name}
+  annotations:${annotations.map(ann => `\n    ${ann}: value`).join('')}
```

Comment:
To be honest I don't think we should list the existing annotations. For many catalogs it'll make the example extremely verbose 😅 It'll also make this logic quite a lot simpler

Reply:
This doesn't list all existing annotations, only the ones passed down in the `props` for `MissingAnnotationEmptyState` component. In most cases, that will just be one annotation.

As an example:

This would render
```ts
<MissingAnnotationEmptyState
    annotation=["db-type", "db-version"]
    readMoreUrl="/docs/default/component/backstage"
  />
```
<img width="372" alt="image" src="https://github.com/backstage/backstage/assets/16994570/c3711853-e19c-424a-b78e-37d42aae708b">

Reply:
@Rugvip is this explanation ok?

Reply:
Yep sounds good! 👍